client/utils/auth.py

```diff
@@ -215,15 +226,26 @@ def log_message(self, format, *args):
             return
 
     server_address = (host_name, port_number)
-    httpd = http.server.HTTPServer(server_address, CodeHandler)
-    httpd.handle_request()
 
-    update_storage(access_token, expires_in, refresh_token)
-    return access_token
+    try:
+        httpd = http.server.HTTPServer(server_address, CodeHandler)
+        httpd.handle_request()
+    except OSError as e:
+        log.warning("HTTP Server Err {}".format(server_address), exc_info=True)
+        raise
+
+    if not auth_error:
+        update_storage(access_token, expires_in, refresh_token)
+        return access_token
+    else:
+        print("Authentication error: {}".format(auth_error))
```

Comment:
It seems like this says that ok can continue to run if there's an authentication error, is that correct? Just want to confirm that this is for allowing students to continue to run tests even if they're having trouble authenticating.

Would returning None cause any issues with other code that assumes that if an access token is returned that it's a string? For example, does the backup endpoint allow the access token to be null? Same with general API requests in network.py.